cmd/plugin/util/util.go

```diff
@@ -101,6 +102,25 @@ func isVersionLessThan(a, b string) bool {
 	return aPatch < bPatch
 }
 
+// PodInDeployment returns whether a pod is part of a deployment with the given name
+// a pod is considered to be in {deployment} if it is owned by a replicaset with a name of format {deployment}-otherchars
+func PodInDeployment(pod apiv1.Pod, deployment string) bool {
+	for _, owner := range pod.OwnerReferences {
+		if owner.Controller == nil || !*owner.Controller || owner.Kind != "ReplicaSet" {
+			continue
+		}
+
+		if strings.Count(owner.Name, "-") != strings.Count(deployment, "-")+1 {
+			continue
+		}
+
+		if strings.HasPrefix(owner.Name, deployment+"-") {
+			return true
+		}
+	}
```

Comment:
please extract the checks into their own function with name that represents what it actually checks. Or add inline comments.

Reply:
nevermind the comment above is explanatory enough